pkg/controller/cronjob/cronjob_controllerv2_test.go

```diff
@@ -1650,23 +1650,57 @@ func TestControllerV2GetJobsToBeReconciled(t *testing.T) {
-					OwnerReferences: []metav1.OwnerReference{{Name: "another-cronjob", Controller: &trueRef}}},
+					OwnerReferences: []metav1.OwnerReference{{Name: "another-cronjob", Controller: &trueRef, Kind: "CronJob"}}},
 				},
 			},
 			expected: []*batchv1.Job{{
 				ObjectMeta: metav1.ObjectMeta{
 					Namespace:       "foo-ns",
 					Name:            "foo-fooer-owner-ref",
 					Labels:          map[string]string{"key": "different-value"},
-					OwnerReferences: []metav1.OwnerReference{{Name: "fooer", Controller: &trueRef}}},
+					OwnerReferences: []metav1.OwnerReference{{Name: "fooer", Controller: &trueRef, Kind: "CronJob"}}},
 			}},
 		},
+		{
+			name:    "test getting jobs in namespace with a controller reference and if kind is present, should be CronJob",
+			cronJob: &batchv1.CronJob{ObjectMeta: metav1.ObjectMeta{Namespace: "foo-ns", Name: "fooer", UID: "fooer-uid-1"}},
+			jobs: []runtime.Object{
+				&batchv1.Job{ObjectMeta: metav1.ObjectMeta{Name: "foo", Namespace: "foo-ns"}},
+				&batchv1.Job{ObjectMeta: metav1.ObjectMeta{Name: "foo-with-owner-ref", Namespace: "foo-ns",
+					OwnerReferences: []metav1.OwnerReference{{Name: "fooer", Controller: &trueRef}}}},
+				&batchv1.Job{ObjectMeta: metav1.ObjectMeta{Name: "foo-with-owner-ref", Namespace: "foo-ns",
+					OwnerReferences: []metav1.OwnerReference{{Name: "fooer-with-dff-name-than-cronjob", Controller: &trueRef, Kind: "CronJob", UID: "fooer-uid-1"}}}},
+				&batchv1.Job{ObjectMeta: metav1.ObjectMeta{Name: "foo-with-diff-owner-ref", Namespace: "foo-ns",
+					OwnerReferences: []metav1.OwnerReference{{Name: "fooer", Controller: &trueRef, Kind: "CustomController"}}}},
+				&batchv1.Job{ObjectMeta: metav1.ObjectMeta{Name: "foo-with-owner-ref-cronjob", Namespace: "foo-ns",
+					OwnerReferences: []metav1.OwnerReference{{Name: "fooer", Controller: &trueRef, Kind: "CronJob", UID: "fooer-uid-1"}}}},
+				&batchv1.Job{ObjectMeta: metav1.ObjectMeta{Name: "foo4", Namespace: "foo-ns"}},
+			},
+			expected: []*batchv1.Job{
+				{ObjectMeta: metav1.ObjectMeta{Name: "foo-with-owner-ref-cronjob", Namespace: "foo-ns",
+					OwnerReferences: []metav1.OwnerReference{{Name: "fooer", Controller: &trueRef, Kind: "CronJob", UID: "fooer-uid-1"}}}},
+			},
+		},
+		{
+			name:    "test getting jobs in namespace with a controller reference and kind is CronJob but UID is different",
+			cronJob: &batchv1.CronJob{ObjectMeta: metav1.ObjectMeta{Namespace: "foo-ns", Name: "fooer", UID: "fooer-uid-1"}},
+			jobs: []runtime.Object{
+				&batchv1.Job{ObjectMeta: metav1.ObjectMeta{Name: "foo-with-owner-ref-cronjob-same-uid", Namespace: "foo-ns",
+					OwnerReferences: []metav1.OwnerReference{{Name: "fooer", Controller: &trueRef, Kind: "CronJob", UID: "fooer-uid-1"}}}},
+				&batchv1.Job{ObjectMeta: metav1.ObjectMeta{Name: "foo-with-owner-ref-cronjob-diff-uid", Namespace: "foo-ns",
+					OwnerReferences: []metav1.OwnerReference{{Name: "fooer", Controller: &trueRef, Kind: "CronJob", UID: "fooer-uid-2"}}}},
```

Comment:
This is testing a stale reference. But we need a test for two or more different jobs and cronjobs and see the correct filtering.